src/ethereum/AddressInfo/index.tsx

```diff
@@ -0,0 +1,87 @@
+import React from 'react';
+import styled from 'styled-components';
+
+import {
+  Icon,
+  Text,
+  Identicon,
+  EllipsisMenu,
+  EllipsisMenuItem,
+  EtherscanButton,
+} from '../../';
+import { textShortener } from '../../utils/strings';
+
+const StyledContainer = styled.div`
+  display: flex;
+`;
+
+const IdenticonContainer = styled.div`
+  margin: 4px 8px 0 0;
+`;
+
+const InfoContainer = styled.div`
+  display: flex;
+  align-items: flex-start;
+  justify-content: center;
+  flex-direction: column;
+`;
+
+const AddressContainer = styled.div`
+  display: flex;
+  align-items: center;
+
+  *:not(:first-child) {
+    margin-left: 12px;
+  }
+`;
+
+type Props = {
+  address: string;
+  shortenAddress?: number;
+  className?: string;
+  name?: string;
+  showIdenticon?: boolean;
+  showCopy?: boolean;
+  menuItems?: EllipsisMenuItem[];
+  showEtherscan?: boolean;
+};
+
+const getShortAddress = (text: string, shortenAddress: number) =>
```

Comment:
I think this wrapper over wrapper is unnecessary, you could use textShortener just fine